PR - ⭐ : Add aks feature

venkatsunilm opened the pull request:
1. Add aks feature
2. Modified readme file
3. Add few changes to common and core classes as well
4. Renamed RG, storage and key valult names to maintain consitency
5. Add documentation
6. Follow few best practices

AymenSegni commented on tf/aks/README.md:
```diff
@@ -0,0 +1,85 @@
+    "name": "ve...@hotmail.com",
+    "type": "user"
+  }
+}
```
Please don't add your tenant details (IDs) into git files, you can change it with dummy values XXXXXXXXXX

venkatsunilm replied:
These are actually dummy values 😊, but yes i will replace them like you suggested.

venkatsunilm replied:
Replaced with dummy names.

AymenSegni commented on tf/aks/README.md:
```diff
@@ -0,0 +1,85 @@
+  "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+  "galleryEndpointUrl": "https://gallery.azure.com/",
+  "managementEndpointUrl": "https://management.core.windows.net/"
+}
```
It is crucial that you do not share your Azure SP credentials, including IDs and secrets, in any git files. This information should be kept private to ensure the security of your account and data. Please replace them with dummy values

venkatsunilm replied:
Sure again it's dummy values but then yes I understood it's better to replace with XXXXXXX. I will replace it. :)

venkatsunilm replied:
Replaced with dummy names and removed the unwanted info.

AymenSegni commented on tf/aks/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+output "aks_client_key" {
+  value       = module.aks.aks_client_key
+  description = "The client key used for authenticating with the AKS cluster."
+}
```
This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

AymenSegni commented on tf/aks/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+output "aks_client_certificate" {
+  value       = module.aks.aks_client_certificate
+  description = "The client certificate used for authenticating with the AKS cluster."
+}
```
This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

AymenSegni commented on tf/aks/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+output "aks_cluster_ca_certificate" {
+  value       = module.aks.aks_cluster_ca_certificate
+  description = "The CA (Certificate Authority) certificate used by the AKS cluster."
+}
```
This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

AymenSegni commented on tf/aks/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+  description = "The CA (Certificate Authority) certificate used by the AKS cluster."
+}
+
+output "aks_cluster_username" {
```
This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

AymenSegni commented on tf/aks/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+output "aks_cluster_password" {
+  value       = module.aks.aks_cluster_password
+  description = "The password for authenticating with the AKS cluster."
+}
```
This a sensitive output that has to be redacted from the Terraform logs using: sensitive   = true
https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs#redact-sensitive-outputs

AymenSegni commented on tf/aks/variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "environment" {
+  description = "Name of the environment"
+  type        = string
+}
```
default is missing

AymenSegni commented on tf/aks/variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "location" {
+  description = "Azure location to use"
+  type        = string
+}
```
default is missing

AymenSegni commented on tf/aks/variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "kubernetes_version" {
+  description = "The Kubernetes version to use"
+  type        = string
+}
```
default is missing

AymenSegni commented on tf/aks/variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "service_principal_client_id" {
+  description = "The client ID of the service principal to be used by AKS"
+  type        = string
+}
```
default is missing

AymenSegni commented on tf/aks/variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "service_principal_client_secret" {
+  description = "The client secret of the service principal to be used by AKS"
+  type        = string
+}
```
default is missing

AymenSegni commented on tf/aks/variables.tf:
```diff
@@ -0,0 +1,29 @@
+
+variable "service_principal_client_secret" {
+  description = "The client secret of the service principal to be used by AKS"
+  type        = string
```
add sensitive = true

AymenSegni commented on tf/aks/variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "ssh_public_key" {
+  description = "The SSH public key to use with AKS"
+  type        = string
+}
```
add sensitive = true

venkatsunilm commented on tf/aks/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+output "aks_host" {
+  value       = module.aks.aks_host
+  description = "The host URL of the AKS cluster."
+}
```
@AymenSegni Is this config and host are sensitive information?

AymenSegni replied:
From the description: " description = "The password for authenticating with the AKS cluster."
It's very sensitive : )

venkatsunilm replied:
I was asking about the config and host not the password. :)

AymenSegni replied:
The host URL isn't sensitive, however config is

venkatsunilm committed:
AKS module: Add aks module, documentation and apply best practices

1. Modified the code for better maintainability and to follow some good practices

2. add documentation, and folloedw some good practices for all three modules including aks

3. More changes to the documentation and missed best practices
